### PR TITLE
Add TableMetadataService and utils

### PR DESCRIPTION
**Pull request by ypeckstadt**
## Description

This PR adds a service and utils that deal with ScalarDB TableMetada. This service and utils will be used for both export and import processes. 

## Related issues and/or PRs

This PR depends on the following PRs and should be reviewed and merged after:

- #1771 

## Changes made

- Added the TableMetadataService
- Added the TableMetadataUtils
- Added the exceptions and error messages that are thrown in the service
- Added unit tests

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes.
- [ ] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

NA

## Release notes

NA

### CODE REVIEW

File: data-loader/core/src/main/java/com/scalar/db/dataloader/core/util/TableMetadataUtils.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  /**
+   * Check if the field is a ScalarDB transaction metadata column or not
+   *
+   * @param columnName Table column name
+   * @param metadataColumns Fixed list of metadata columns
+   * @param columnNames List of all column names in a table
+   * @return The field is metadata or not
+   */
+  public static boolean isMetadataColumn(
```
Comment (brfrn169) [inline data-loader/core/src/main/java/com/scalar/db/dataloader/core/util/TableMetadataUtils.java:33]: We might be able to use [ConsensusCommitUtils](https://github.com/scalar-labs/scalardb/blob/03f7321282d0e335f333197af2a0a59b07f12c21/core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitUtils.java). Could you please check it?

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 2]
```diff
@@ -610,6 +610,16 @@ public enum CoreError implements ScalarDbError {
       "Invalid file extension: %s. Allowed extensions are: %s",
       "",
       ""),
+  DATA_LOADER_INVALID_COLUMN_KEY_PARSING_FAILED(
+      Category.USER_ERROR, "0136", "Invalid key: Column %s does not exist in the table.", "", ""),
+  DATA_LOADER_INVALID_VALUE_KEY_PARSING_FAILED(
+      Category.USER_ERROR, "0137", "Parsing of key value %s failed. Details:%s.", "", ""),
```
Comment (Torch3333) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:616]:
```suggestion
      Category.USER_ERROR, "0137", "Parsing of key value %s failed. Details: %s.", "", ""),
```

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!
I just have a minor comment.

**Comment by ypeckstadt** [comment]
Converting to draft PR because, as Suzuki san mentioned, this PR might not be necessary.

**Comment by ypeckstadt** [comment]
Closing this PR as things have changed a bit and need to update the code to use the ConsensusCommitUtils instead.

**Pull request closed without merging**